### PR TITLE
feat: Improve quote handling in lexer

### PR DESCRIPTION
**Pull request by max-sixty**
Improvements described inline

### CODE REVIEW

**Comment by max-sixty** [comment]
Chumsky errors were actually decipherable once I got used to them!

File: crates/prql-parser/src/lexer.rs [hunk 1]
```diff
@@ -387,3 +390,26 @@ impl std::hash::Hash for Token {
...
+    // sure this is correct — possibly the first six characetrs should be a
+    // closed string?
+    assert_snapshot!(quoted_string(false).parse(r#"''''''aoeu''''''"#).unwrap(), @"'aoeu");
+
+    // When not escaping, we take the inner string between the three quotes
+    assert_snapshot!(quoted_string(false).parse(r#""""\"hello\""""#).unwrap(), @r###"\"hello\"###);
+
+    // Escape each inner quote depending on the outer quote
+    assert_snapshot!(quoted_string(true).parse(r#""\"hello\"""#).unwrap(), @r###""hello""###);
+    assert_snapshot!(quoted_string(true).parse(r#"'\'hello\''"#).unwrap(), @"'hello'");
```
Comment (max-sixty) [inline crates/prql-parser/src/lexer.rs:416]: This was previously handled incorrectly, IIUC

**Comment by max-sixty** [comment]
@aljazerzen the gentlest of pings, no real rush

**Review by aljazerzen — APPROVED** [review]
Much better and faster probably. And it handles arbitrary many opening quotes.

**Pull request merged**